firefox-bootstrap/bootstrap.js

```diff
@@ -0,0 +1,259 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+let { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "CustomizableUI",
+  "resource:///modules/CustomizableUI.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+const PREF_WC_REPORTER_ENABLED = "extensions.webcompat-reporter.enabled";
+const WEBCOMPAT_DOMAIN = "webcompat.com";
+const WIDGET_ID = "webcompat-reporter-button";
+
+class TabListener {
+  constructor(browser) {
+    this.browser = browser;
+    this.addListeners();
+  }
+
+  get QueryInterface() {
+    return XPCOMUtils.generateQI(["nsIWebProgressListener",
+      "nsISupportsWeakReference"]);
+  }
+
+  addListeners() {
+    this.browser.addProgressListener(this);
+    this.browser.tabContainer.addEventListener("TabSelect",
+      this.setButtonState);
+  }
+
+  removeListeners() {
+    this.browser.removeProgressListener(this);
+    this.browser.tabContainer.removeEventListener("TabSelect",
+      this.setButtonState);
+  }
+
+  onStateChange(webProgress, request, flag, status) {
+    if (flag & Ci.nsIWebProgressListener.STATE_STOP) {
+      // about:new-tab loads don't have an originalURI property, but
+      // have a request.name of about:document-onload-blocker
+      let scheme = request.originalURI ? request.originalURI.scheme
+                 : request.name;
+      this.setButtonState(scheme);
+    }
+  }
+
+  setButtonState(arg) {
+    // Did we get here from a TabSelect event (arg = event object)
+    // or manually via onStateChange (arg = string)?
+    let scheme = typeof arg === "string" ? arg :
+                 arg.target.linkedBrowser.currentURI.scheme;
+    let button = CustomizableUI.getWidget(WIDGET_ID);
+    let isInvalidScheme = ["about", "chrome", "file", "resource", "view-source"].
+      some((prefix) => scheme.startsWith(prefix));
+
+    button.disabled = isInvalidScheme;
+  }
+}
+
+let windowListener = new class {
+  onOpenWindow(window) {
+    let win = window.QueryInterface(Ci.nsIInterfaceRequestor). // eslint-disable-line new-cap
+      getInterface(Ci.nsIDOMWindow);
+    let browser;
+
+    win.addEventListener("load", function onLoad() {
+      win.removeEventListener("load", onLoad, false);
+      browser = win.gBrowser;
+      browser.tabListener = new TabListener(browser);
+
+      win.addEventListener("unload", function onUnLoad(e) {
+        win.removeEventListener("unload", onUnLoad);
+        browser.tabListener.removeListeners();
+        browser.tabListener = null;
+      });
+    });
+  }
+};
+
+let httpRequestObserver = new class {
+  constructor() {
+    this.TOPIC = "http-on-modify-request";
+    this.addObserver();
```

Comment:
This is still valid. If you have the add-on installed and manually report an issue on webcompat.com, the `X-Reported-With` is set. I don't think that's what you want. You call this function in line 225 for the webcompat.com tab opened for reporting the issue via add-on.

Unless you want to set the header for all issues that get reported, including those reported by people manually accessing webcompat.com, this call not needed here.